plugin/externalPluginEntry.go

```diff
@@ -80,56 +75,63 @@ func (e *externalPluginEntry) setCacheTTLs(ttls decodedCacheTTLs) {
 	}
 }
 
-// List lists the entry's children, if it has any.
+// implements returns true if the entry implements the given method,
+// false otherwise
+func (e *externalPluginEntry) implements(method string) bool {
+	for _, m := range e.methods {
+		if method == m {
+			return true
+		}
+	}
+	return false
+}
+
 func (e *externalPluginEntry) List(ctx context.Context) ([]Entry, error) {
-	stdout, err := e.script.InvokeAndWait(ctx, "list", e.id(), e.state)
+	stdout, err := e.script.InvokeAndWait(ctx, "list", e)
 	if err != nil {
 		return nil, err
 	}
-
 	var decodedEntries []decodedExternalPluginEntry
 	if err := json.Unmarshal(stdout, &decodedEntries); err != nil {
 		activity.Record(
 			ctx,
 			"could not decode the entries from stdout\nreceived:\n%v\nexpected something like:\n%v",
 			strings.TrimSpace(string(stdout)),
-			"[{\"name\":\"<name_of_first_entry>\",\"supported_actions\":[\"list\"]},{\"name\":\"<name_of_second_entry>\",\"supported_actions\":[\"list\"]}]",
+			"[{\"name\":\"<name_of_first_entry>\",\"wmethods\":[\"list\"]},{\"name\":\"<name_of_second_entry>\",\"wmethods\":[\"list\"]}]",
```

Comment:
Ah yeah. I might as well add tests for these cases. Didn't at the time b/c I didn't want to deal with a long regex match.

Reply:
Nevermind, there isn't a good way to test this since it is documentation (meaning the assertion is the example against itself), and the activity package isn't mocked. I'll just have to be more diligent in updating them.

I could make these internal variables that consist of serialized decoded external plugin entries (so that a change like `wmethods` would throw a compiler error), but that's not worth the complexity right now. May look into it later.